src/server/system_services/account_server.js

```diff
@@ -654,10 +654,13 @@ function delete_account_by_property(req) {
             if (!is_support_or_admin_or_me(req.system, req.account, account_to_delete)) {
                 throw new RpcError('UNAUTHORIZED', 'Cannot delete account');
             }
-            if (system_store.data.buckets.find(b => String(b.owner_account) === String(account_to_delete._id))) {
-                throw new RpcError('UNAUTHORIZED', 'Cannot delete account that is owner of buckets');
-            }
 
+            for (const bucket of system_store.data.buckets) {
```

Comment:
there is any change here except of the UNAUTHORIZED -> FORBIDDEN?
why can't we use the same logic? I also think this can maybe be a function called check_buckets_owner or something...

Reply:
yes I patched an old fix for delete_account() that needs to be also on delete_account_by_property()
moved the entire verification for both functions to a united function